src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/scarlet_monastery/instance_scarlet_monastery.cpp

```diff
@@ -40,15 +40,20 @@ void instance_scarlet_monastery::OnCreatureCreate(Creature* pCreature)
     switch (pCreature->GetEntry())
     {
         case NPC_MOGRAINE:
-            m_npcEntryGuidStore[pCreature->GetEntry()] = pCreature->GetObjectGuid();
             if (GetData(TYPE_ASHBRINGER_EVENT) == IN_PROGRESS)
                 DoOrSimulateScriptTextForThisInstance(SAY_ASHBRINGER_ENTRANCE, NPC_MOGRAINE);
```

Comment:
This is never executed for some reason. Guessing 2 reasons: 1) The Dialogue isn't implemented and/or 2) Mograine doesn't spawn on instance enter, so this code only gets called later when he is eventually spawned